File: ios/Awaytime/Testing/FeatureParityValidator.swift
import SwiftUI
import XCTest

/// Validates feature parity between iOS and Android implementations
class FeatureParityValidator {
    
    static let shared = FeatureParityValidator()
    
    private init() {}
    
    // MARK: - Core Feature Validation
    
    func validateUsageTracking() -> ValidationResult {
        var issues: [String] = []
        
        // Check data structure consistency
        let usageService = UsageTrackingService()
        
        Task {
            do {
                let usageData = try await usageService.getCurrentUsage()
                
                for usage in usageData {
                    if usage.appIdentifier.isEmpty {
                        issues.append("Usage data missing app identifier")
                    }
                    if usage.totalTime < 0 {
                        issues.append("Usage data has negative time")
                    }
                }
            } catch {
                issues.append("Usage tracking failed: \(error)")
            }
        }
        
        return ValidationResult(
            feature: "Usage Tracking",
            isValid: issues.isEmpty,
            issues: issues
        )
    }
    
    func validateAppBlocking() -> ValidationResult {
        var issues: [String] = []
        
        let blockingService = AppBlockingService()
        let testApps = ["com.example.test"]
        
        Task {
            do {
                try await blockingService.blockApps(testApps)
                let blockedApps = await blockingService.getCurrentlyBlockedApps()
                
                if !blockedApps.contains(testApps[0]) {
                    issues.append("App blocking not working correctly")
                }
                
                try await blockingService.unblockApps(testApps)
            } catch {
                issues.append("App blocking failed: \(error)")
            }
        }
        
        return ValidationResult(
            feature: "App Blocking",
            isValid: issues.isEmpty,
            issues: issues
        )
    }
    
    func validateColorScheme() -> ValidationResult {
        var issues: [String] = []
        
        // Validate purple color consistency
        let primaryColor = Color.purple
        
        // Check if color matches expected hex value
        // In a real implementation, you'd extract and compare color values
        
        return ValidationResult(
            feature: "Color Scheme",
            isValid: issues.isEmpty,
            issues: issues
        )
    }
    
    // MARK: - Comprehensive Validation
    
    func validateAllFeatures() -> [ValidationResult] {
        return [
            validateUsageTracking(),
            validateAppBlocking(),
            validateColorScheme()
        ]
    }
}

struct ValidationResult {
    let feature: String
    let isValid: Bool
    let issues: [String]
    
    var summary: String {
        if isValid {
            return "✅ \(feature): All checks passed"
        } else {
            return "❌ \(feature): \(issues.count) issues found"
        }
    }
}

File: android/app/src/main/java/com/awaytime/app/testing/FeatureParityValidator.kt
package com.awaytime.app.testing

import android.content.Context
import com.awaytime.app.service.AppBlockingService
import com.awaytime.app.service.UsageTrackingService
 import com.awaytime.app.service.getCurrentlyBlockedApps
import com.awaytime.app.ui.theme.AwayTimeColors
import kotlinx.coroutines.runBlocking

/**
 * Validates feature parity between Android and iOS implementations
 */
class FeatureParityValidator(private val context: Context) {

    // MARK: - Core Feature Validation

    suspend fun validateUsageTracking(): ValidationResult {
        val issues = mutableListOf<String>()

        try {
            // Check data structure consistency
            val usageService = UsageTrackingService(context)
            val usageData = usageService.getCurrentUsage()

            for (usage in usageData) {
                if (usage.appIdentifier.isEmpty()) {
                    issues.add("Usage data missing app identifier")
                }
                if (usage.totalTime < 0) {
                    issues.add("Usage data has negative time")
                }
            }
        } catch (e: Exception) {
            issues.add("Usage tracking failed: ${e.message}")
        }

        return ValidationResult(
            feature = "Usage Tracking",
            isValid = issues.isEmpty(),
            issues = issues
        )
    }

    fun validateAppBlocking(): ValidationResult {
        val issues = mutableListOf<String>()

        try {
            val blockingService = AppBlockingService(context)
            val testApps = listOf("com.example.test")

            runBlocking {
                blockingService.blockApps(testApps)
                val blockedApps = blockingService.getCurrentlyBlockedApps()

                if (!blockedApps.contains(testApps[0])) {
                    issues.add("App blocking not working correctly")
                }

                blockingService.unblockApps(testApps)
            }
        } catch (e: Exception) {
            issues.add("App blocking failed: ${e.message}")
        }

        return ValidationResult(
            feature = "App Blocking",
            isValid = issues.isEmpty(),
            issues = issues
        )
    }

    fun validateColorScheme(): ValidationResult {
        val issues = mutableListOf<String>()

        // Validate purple color consistency
        val primaryColor = AwayTimeColors.primary

        // Check if color matches expected value
        // In a real implementation, you'd compare color values

        return ValidationResult(
            feature = "Color Scheme",
            isValid = issues.isEmpty(),
            issues = issues
        )
    }

    // MARK: - Comprehensive Validation

    suspend fun validateAllFeatures(): List<ValidationResult> {
        return listOf(
            validateUsageTracking(),
            validateAppBlocking(),
            validateColorScheme()
        )
    }

    companion object {
        fun create(context: Context): FeatureParityValidator {
            return FeatureParityValidator(context)
        }
    }
}

data class ValidationResult(
    val feature: String,
    val isValid: Boolean,
    val issues: List<String>
) {
    val summary: String
        get() = if (isValid) {
            "✅ $feature: All checks passed"
        } else {
            "❌ $feature: ${issues.size} issues found"
        }
}